pypsa/io.py

```diff
@@ -870,8 +877,14 @@ def import_components_from_dataframe(network, dataframe, cls_name):
         else:
             if static_attrs.at[k, "type"] == "string":
                 dataframe[k] = dataframe[k].replace({np.nan: ""})
-
-            dataframe[k] = dataframe[k].astype(static_attrs.at[k, "typ"])
+            if dataframe[k].dtype != static_attrs.at[k, "typ"]:
+                if static_attrs.at[k, "type"] == "geometry":
+                    crs = network.meta.pop("_crs", None)
```

Comment:
Why do we need to go through `n.meta` to retrieve the CRS? Couldn't we just store `n.crs` with the exported files?

Reply:
The `crs` attribute is actually a network property which need some data reference, I found this the cleanest solution (alternatively, one could create an intermediate n._crs object, which then also has to be deleted after loading (if it is there))

Reply:
This should be possible to do analogous to `n.srid`, right?

Reply:
Not really, since `n.crs` is a `property` which does not have a setter.

Reply:
Ok, if it's difficult, I am fine with this!